lib/queue.rb

```diff
@@ -1,28 +1,38 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new
+    @front = 0 
+    @back = 0 
```

Comment:
If you use a circular buffer, you need something to indicate when the Queue is empty. (-1 works)

```suggestion
    @front = -1
    @back = -1
```